dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java

```diff
@@ -655,10 +655,18 @@ public void save(Permission permission, Permissionable permissionable, User user
 	 */
 	@WrapInTransaction
 	private void save(Permission permission, Permissionable permissionable, User user, boolean respectFrontendRoles, boolean createEvent) throws DotDataException, DotSecurityException {
-		if(!doesUserHavePermission(permissionable, PermissionAPI.PERMISSION_EDIT_PERMISSIONS, user)) {
+		boolean isCopy = false;
+		if (permissionable instanceof Contentlet){

```

Comment:
I would encapsulate this into a method isCopy()
this methods are already so big